### DISCUSSION TITLE
xarray.Dataset.from_array

### DISCUSSION THREAD

**Discussion by raybellwaves** [pydata/xarray#5768] [Ideas] [answered]
I would be interested in a `xarray.Dataset.from_numpy` method.

This would act as a wrapper for `xr.DataArray(arr).to_dataset("arr")` or a better default variable name than "arr".

Motivation here is a one-line inverse transform of `ds.to_numpy()`.

Obviously this has draw backs compared to building up the xarray object using `xr.DataArray` and using its args. However, I feel would speed up the process of a using Dataset operations on a numpy array.

**Comment by TomNicholas** [comment, marked as answer]
Hi @raybellwaves, thanks for the suggestion.

> Motivation here is a one-line inverse transform of ds.to_numpy().

I'm a bit confused though - this is already a one-line conversion? You could of course make it neater by locally defining

```python
def numpy_array_to_ds(arr, name="arr"):
    return xr.DataArray(arr).to_dataset(name=name)
```

I suppose we could perhaps try to further generalise `Dataset.__init__`, although it is already quite a complicated constructor...

> Motivation here is a one-line inverse transform of ds.to_numpy().

(FYI the original motivation for `ds.to_numpy()` was situations where you need to strip the array of any other duck-type wrappers, for instance pint units. I feel like that isn't really related to what you're asked for here though - what you're asking about is closer to an inverse of `.values()`.)

**Reply by raybellwaves** [reply]
Thanks for explaining the motivation/purpose of `.to_numpy()` @TomNicholas. Yeah I probably want `xarray.Dataset.from_array` as the inverse of `ds[key].values`. Motivation of the question was when I was going through xarray data format transformations e.g. `to_dataframe` and `from_dataframe`. There probably a balance between creating new methods which are simplified chained methods of the current API. Not high priority.

**Reply by raybellwaves** [reply]
Similar to https://github.com/pydata/xarray/discussions/5795#discussioncomment-1334694 I agree it's best not duplicate existing methods for future maintenance.
